src/clients/python/experimental_api_v2/library/grpcclient.py

```diff
@@ -1077,36 +1068,39 @@ def set_data_from_numpy(self, input_tensor):
         else:
             self._input.contents.raw_contents = input_tensor.tobytes()
 
-    def set_parameter(self, key, value):
-        """Adds the specified key-value pair in the requested input parameters
+    def set_data_from_shared_memory(self, region_name, byte_size, shape,
+                                    datatype):
+        """Set the tensor data from the specified shared memory region.
 
         Parameters
         ----------
-        key : str
-            The name of the parameter to be included in the request. 
-        value : str/int/bool
-            The value of the parameter
+        region_name : str
+            The name of the shared memory region holding tensor data.
+        byte_size : int
+            The size of the shared memory region holding tensor data.
+        shape : list
+            The shape of the associated tensor input.
+        datatype : str
+            The datatype of the associated tensor input.
         
         """
-        if not type(key) is str:
-            raise_error(
-                "only string data type for key is supported in parameters")
-
-        param = self._input.parameters[key]
-        if type(value) is int:
-            param.int64_param = value
-        elif type(value) is bool:
-            param.bool_param = value
-        elif type(value) is str:
-            param.string_param = value
-        else:
-            raise_error("unsupported value type for the parameter")
 
-    def clear_parameters(self):
-        """Clears all the parameters that have been added to the input request.
+        self._input.parameters[
+            'shared_memory_region'].string_param = region_name
+        self._input.parameters[
+            'shared_memory_byte_size'].int64_param = byte_size
+
+        self._input.ClearField('shape')
+        self._input.shape.extend(shape)
+        self._input.datatype = datatype
+
+    def reset(self):
```

Comment:
Why a reset function? If they want to reuse the request object can't they just call use_shared_memory again with different values? If they want bigger changes (like changing the shape), then they just have to create a new input from scratch.